src/backend/distributed/planner/multi_logical_planner.c

```diff
@@ -676,41 +746,20 @@ DeferErrorIfCannotPushdownSubquery(Query *subqueryTree, bool outerMostQueryHasLi
 							 errorDetail, NULL);
 	}
 
-	/* recursively do same check for subqueries of this query */
```

Comment:
If we make this function non-recursive, we should also update the function comments.